hal/mbed_sleep_manager.c

```diff
@@ -91,14 +91,18 @@ static sleep_statistic_t* sleep_tracker_add(const char* const filename)
         }
     }
 
-    debug("No free indexes left to use in mbed sleep tracker.\r\n");
+    MBED_ASSERT(false && "No free indexes left to use in mbed sleep tracker");
 
     return NULL;
 }
 
 static void sleep_tracker_print_stats(void)
 {
-    debug("Sleep locks held:\r\n");
+    if (core_util_in_critical_section()) {
```

Comment:
`core_util_in_critical_section()` - determines on if we are in critical section or not. 
We might be in ISR and not in critical section, I guess additional check for ISR will be needed with `core_util_is_isr_active()`